monitor/data_monitoring.py

```diff
@@ -93,18 +101,22 @@ def _send_alerts(self, days_back: int):
         if alert_count > 0:
             self._send_to_slack(alerts)
 
-    def run(self, days_back: int, dbt_full_refresh: bool = False) -> None:
+    def run(self, days_back: int, dbt_full_refresh: bool = False) -> int:
 
         logger.info("Running internal dbt run to aggregate alerts")
         success = self.dbt_runner.run(models='alerts', full_refresh=dbt_full_refresh)
         self.execution_properties['alerts_run_success'] = success
         if not success:
             logger.info('Could not aggregate alerts successfully')
-            return
+            self.execution_properties['run_status_code'] = 1
```

Comment:
In case that run will not be the last thing that can change the status_code, it is better to set the status_code and pass it explicit to the execution_properties
```suggestion
            self.status_code = 1
            self.execution_properties['run_status_code'] = self.status_code
```